ASPNetCoreMastersTodoList.Api/Filters/EnsureItemExistsFilter.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Services;
+
+namespace ASPNetCoreMastersTodoList.Api.Filters
+{
+    public class EnsureItemExistsFilter : IActionFilter
+    {
+        private readonly ItemExistService _service;
+
+        public EnsureItemExistsFilter(ItemExistService service)
+        {
+            _service = service;
+        }
+        public void OnActionExecuted(ActionExecutedContext context)
+        {
+            
+        }
+
+        public void OnActionExecuting(ActionExecutingContext context)
+        {
+            var itemId = (int)context.ActionArguments["itemId"];
```

Comment:
This will throw an exception if itemId is not available in the request arguments, make sure to check first if arguments contains itemID, Thanks!